src/Dotnet.Script.DependencyModel/Logging/LogActionExtensions.cs

```
@@ -1,22 +1,69 @@
         public static void Debug(this Logger logger, string message) => logger(LogLevel.Debug, message);
-        public static void Info(this Logger logger, string message) => logger(LogLevel.Info, message);       
+        public static void Info(this Logger logger, string message) => logger(LogLevel.Info, message);
+        public static void Warning(this Logger logger, string message) => logger(LogLevel.Warning, message);
+        public static void Error(this Logger logger, string message, Exception exception = null) => logger(LogLevel.Error, message, exception);
+        public static void Critical(this Logger logger, string message, Exception exception = null) => logger(LogLevel.Critical, message, exception);
+    }
+
+    public static class LevelMapper
+    {
+        private static Dictionary<string, LogLevel> _levelMap = CreateMap();
+
+        private static Dictionary<string, LogLevel> CreateMap()
+        {
+            var map = new Dictionary<string, LogLevel>(StringComparer.InvariantCultureIgnoreCase);
+            map.Add("t", LogLevel.Trace);
+            map.Add("trace", LogLevel.Trace);
+            map.Add("d", LogLevel.Debug);
+            map.Add("debug", LogLevel.Debug);
+            map.Add("i", LogLevel.Info);
+            map.Add("info", LogLevel.Info);
+            map.Add("w", LogLevel.Warning);
+            map.Add("warning", LogLevel.Warning);
+            map.Add("e", LogLevel.Error);
+            map.Add("error", LogLevel.Error);
+            map.Add("c", LogLevel.Critical);
+            map.Add("critical", LogLevel.Critical);
+            return map;
+        }
+
+        public static LogLevel FromString(string levelName)
+        {
+            if (string.IsNullOrWhiteSpace(levelName))
+            {
+                return LogLevel.Warning;
+            }
+
+            if (!_levelMap.TryGetValue(levelName, out var level))
+            {
+                throw new InvalidOperationException($"Unknown level name : {levelName}");
```

should this throw or maybe better to silently fallback to some default log level like Info. it would not be great for logging config to crash the app?